src/plugins/intel_cpu/tests/unit/shape_inference_test/custom_shape_infer/matmul.cpp

```diff
@@ -82,63 +84,60 @@ class CPUMatMulTest : public TestWithParam<matmul_test_params_t> {
     unit_test::cpu_test_shape_infer(matmul.get(), static_input_shapes, output_shapes);
 }
 
 TEST_P(CPUMatMulTest, transpose_input_b) {
-    GTEST_SKIP() << "Skipping test, please check CVS-108946";
     const auto matmul = make_matmul(a_shape.size(), b_shape.size(), false, true);
 
     const auto b_transpose = make_transpose_input(b_shape);
     std::vector<StaticShape> static_input_shapes = {a_shape, b_transpose};
 
-    // TODO 108946,below test case can't pass
+    matmul->set_output_type(0, element::i64, ov::PartialShape(std::vector<ov::Dimension>(exp_shape.size(), -1)));
     unit_test::cpu_test_shape_infer(matmul.get(), static_input_shapes, output_shapes);
 }
 
 TEST_P(CPUMatMulTest, transpose_inputs_a_b) {
-    GTEST_SKIP() << "Skipping test, please check CVS-108946";
     const auto matmul = make_matmul(a_shape.size(), b_shape.size(), true, true);
 
     const auto a_transpose = make_transpose_input(a_shape);
     const auto b_transpose = make_transpose_input(b_shape);
 
     std::vector<StaticShape> static_input_shapes = {a_transpose, b_transpose};
 
-    // TODO 108946,below test case can't pass
+    matmul->set_output_type(0, element::i64, ov::PartialShape(std::vector<ov::Dimension>(exp_shape.size(), -1)));
     unit_test::cpu_test_shape_infer(matmul.get(), static_input_shapes, output_shapes);
 }
 
 /** \brief Use transpose order -> output shape dimensions shall be as transpose order. */
 INSTANTIATE_TEST_SUITE_P(CpuShapeInfer,
                          CPUMatMulTest,
+                         // only support rankA = rankB
                          Values(make_tuple(StaticShape({1}), StaticShape({1})),
-                                make_tuple(StaticShape({1}), StaticShape({1, 3})),
-                                make_tuple(StaticShape({1}), StaticShape({1, 1, 3})),
-                                make_tuple(StaticShape({3, 1}), StaticShape({1})),
-                                make_tuple(StaticShape({3, 2, 1}), StaticShape({1})),
                                 make_tuple(StaticShape({3}), StaticShape({3})),
                                 make_tuple(StaticShape({5, 2}), StaticShape({2, 6})),
-                                make_tuple(StaticShape({2, 1, 2}), StaticShape({2, 6})),
                                 make_tuple(StaticShape({10, 8, 9, 2}), StaticShape({10, 8, 2, 8})),
                                 make_tuple(StaticShape({3, 1, 4, 3, 4}), StaticShape({3, 2, 1, 4, 1}))),
+                                // make_tuple(StaticShape({1}), StaticShape({1, 3})),
+                                // make_tuple(StaticShape({1}), StaticShape({1, 1, 3})),
+                                // make_tuple(StaticShape({3, 1}), StaticShape({1})),
+                                // make_tuple(StaticShape({3, 2, 1}), StaticShape({1})),
+                                // make_tuple(StaticShape({2, 1, 2}), StaticShape({2, 6})),
```

Comment:
If such test cases are not supported, and most importantly are not supposed to be supported, then it does make sense to remove them? What do you think?

Reply:
I totally agree with that we should support it, from the comment https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/shape_inference/custom/matmul.cpp#L27 , only some rank cases are implemented. I don't have the background information, so I don't know if need to do it

Reply:
From the comment you mentioned, it follows that for the cases of different ranks we have to fallback on the ngraph implementation. It can be done on the factory level pretty easy.

Reply:
add NgrapShapeInfer branch when rank0 != rank1